### PR TITLE
Implemented compression level algorithms properly

### PR DESCRIPTION
**Pull request by dgquintas**
as a function of the accepted algorithms by the call's peer.

### CODE REVIEW

File: src/core/surface/call.c [hunk 1]
```diff
@@ -1481,3 +1481,9 @@ void *grpc_call_context_get(grpc_call *call, grpc_context_index elem) {
 }
 
 uint8_t grpc_call_is_client(grpc_call *call) { return call->is_client; }
+
+grpc_compression_algorithm grpc_call_compression_for_level(
+    grpc_call *call, grpc_compression_level level) {
+  return grpc_compression_algorithm_for_level(level,
```
Comment (ctiller) [inline src/core/surface/call.c]: Does this need a lock?

**Comment by ctiller** [comment]
One small thing to check, otherwise LGTM

**Comment by ctiller** [comment]
We need a `set_compression_level()` on ClientContext in C++ also

File: src/core/surface/call.c (hunk 1 above)
Comment (dgquintas) [inline src/core/surface/call.c]: Done.

**Comment by dgquintas** [comment]
Greenish

**Pull request merged**